xrpc/xrpc.go

```diff
@@ -112,8 +112,7 @@ func (c *Client) Do(ctx context.Context, kind XRPCRequestType, inpenc string, me
 	defer resp.Body.Close()
 
 	if resp.StatusCode != 200 {
-		var i interface{}
-		_ = json.NewDecoder(resp.Body).Decode(&i)
+		io.Copy(io.Discard, resp.Body)
```

Comment:
We probably want to capture the error out of this. I think theres a standard error format that gets returned here

Reply:
yup, I believe this is still the expected format for XRPC endpoint errors: https://atproto.com/specs/xrpc#custom-error-codes-and-descriptions

currently gateway errors (aka, load balancer can't connect to PDS) will not be proper JSON, and that is likely to be a common issue with PDS deployments/implementations. So we should *try* to parse and log (or otherwise report) error messages, but need to handle HTML coming back, and take just the HTTP status code in that case.